src/com_tjvendors/admin/models/vendors.php

```diff
@@ -208,6 +208,32 @@ public function deleteClientFromVendor($vendor_id,$client)
 		$tjvendorsModelVendor = JModelLegacy::getInstance('Vendor', 'TjvendorsModel');
 		$vendorData           = $tjvendorsModelVendor->getItem($vendor_id);
 		$db                   = $this->getDbo();
+		
+		//Get vendor specific Fees Data
+		$query = $db->getQuery(true);
+		$query->select('*');
+		$query->from($db->quoteName('#__tjvendors_fee'));
+		$query->where($db->quoteName('vendor_id') . ' = ' . (int) $vendor_id);
+		$query->where($db->quoteName('client') . ' = ' . $db->quote($client));
+		$db->setQuery($query);
+		$VendorFeeData = $db->loadAssocList();
+		
+		if (!empty($VendorFeeData))
+		{
+			foreach ($VendorFeeData as $FeeData)
+			{
+				// Getting Vendor Payable Amount here
+				$result = TjvendorsHelper::getPayableAmount($FeeData['vendor_id'], $FeeData['client'], $FeeData['currency']);
```

Comment:
What is this method doing ?

There should be a single method in the vendor model that gets the outstanding amount for a single vendor. That is the method you should be simply calling once, not even in a loop.

Reply:
@coolbung 
Do you mean the function will return the array of 'currency' => payable amount?

Reply:
Resolved